### PR TITLE
Nrfx 6558 implement device info service on secure domain

### PR DESCRIPTION
**Pull request by ayla-nordicsemi**

### CODE REVIEW

File: samples/sdfw/ssf_client/src/main.c [hunk 1]
```diff
@@ -70,6 +71,21 @@ static void sdfw_update(void)
 	}
 }
 
+static void device_info(void)
+{
+	int err;
+
+	char uuid_value[41] = "00000000-0000-0000-0000-000000000000\0";
```
Comment (parttimaa) [inline samples/sdfw/ssf_client/src/main.c]: Should the `41` actually be `36+1`?

File: samples/sdfw/ssf_client/src/main.c [hunk 2]
```diff
@@ -70,6 +71,21 @@ static void sdfw_update(void)
...
+	int uuid_value_len = strlen(uuid_value);
```
Comment (parttimaa) [inline samples/sdfw/ssf_client/src/main.c]: UUID length is always the same. So I see no benefit in getting the length because it always returns `36` here.

File: subsys/sdfw_services/services/device_info/device_info_service.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+enum {
+	UUID = 0,
+	HWVERSION,
+} device_info_target;
+
+enum {
+	SUCCESS = 0,
+	INTERNAL_ERROR = 0x1001001,
+	UNPROGRAMMED = 0x1001002,
```
Comment (parttimaa) [inline subsys/sdfw_services/services/device_info/device_info_service.c]: Is it okay to reuse `0x100_1001` and `0x100_1002` values which are already in use by IMEI service? Or vice versa, do we always want to use the same codes for these?

File: subsys/sdfw_services/services/device_info/device_info_service.c [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	struct device_info_service_read_resp uuid_read_resp = read_resp.device_info_resp_msg_device_info_service_read_resp_m;
+	if (read_resp.device_info_resp_msg_choice == device_info_resp_msg_device_info_service_read_resp_m_c) {
+		if(UUID == uuid_read_resp.device_info_service_read_resp_target) {
+			ret = read_resp.device_info_resp_msg_device_info_service_read_resp_m.device_info_service_read_resp_status;
+			if (SUCCESS != ret) {
+				ssf_client_decode_done(rsp_pkt);
+				return ret;
+			}
+		} else {
+			// not a read UUID response
```
Comment (parttimaa) [inline subsys/sdfw_services/services/device_info/device_info_service.c]: Maybe rephrase `not a read` as `did not read` or `failed to read UUID response` or something like that. Similarly on line #74.

File: subsys/sdfw_services/services/device_info/device_info_service.cddl [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+; struct typedefs
+; .. include_startingpoint_device_info_cddl_rst
+; Device Info service request payload.
+device_info_service_read_req = (
+      2,
+      target: uint, ; 0 - uuid, 1 - hw_version
+)
+
+device_info_service_write_req = (
```
Comment (parttimaa) [inline subsys/sdfw_services/services/device_info/device_info_service.cddl:26]: I assume that these are written in production already before any SD FW is even programmed?

File: subsys/sdfw_services/services/device_info/zcbor_generated/device_info_service_decode.c [hunk 6]
```diff
@@ -0,0 +1,136 @@
+/*
+ * Generated using zcbor version 0.9.0
+ * https://github.com/NordicSemiconductor/zcbor
+ * Generated with a --default-max-qty of 3
+ */
+
```
Comment (parttimaa) [inline subsys/sdfw_services/services/device_info/zcbor_generated/device_info_service_decode.c:12]: Should there be copyright?

File: subsys/sdfw_services/services/device_info/device_info_service.cddl (hunk 5 above)
Comment (ayla-nordicsemi) [reply subsys/sdfw_services/services/device_info/device_info_service.cddl:26]: no it is not the case, this is a data model for the format of data to be exchanged between secure domain server and clients. nothing is written in production. 
this file dictates what is generated in the folder zcbor_generated.

File: subsys/sdfw_services/services/device_info/zcbor_generated/device_info_service_decode.c (hunk 6 above)
Comment (ayla-nordicsemi) [reply subsys/sdfw_services/services/device_info/zcbor_generated/device_info_service_decode.c:12]: done

File: samples/sdfw/ssf_client/src/main.c (hunk 1 above)
Comment (ayla-nordicsemi) [reply samples/sdfw/ssf_client/src/main.c]: this has been removed.

File: samples/sdfw/ssf_client/src/main.c (hunk 2 above)
Comment (ayla-nordicsemi) [reply samples/sdfw/ssf_client/src/main.c]: this has been reworked.

File: subsys/sdfw_services/services/device_info/device_info_service.c (hunk 3 above)
Comment (ayla-nordicsemi) [reply subsys/sdfw_services/services/device_info/device_info_service.c]: this is now generated in an enum based on the cddl data model file.
